satellite/marketing/marketingweb/server.go

```diff
@@ -42,7 +42,25 @@ type Server struct {
 	}
 }
 
-// commonPages returns templates that are required for everything.
+// Struct used to render each offer table
+type offerSet struct {
+	RefOffers,FreeCredits []marketing.Offer
+}
+
+// Organizes offers from database by type.
+func organizeOffers(offers []marketing.Offer) offerSet{
```

Comment:
You can use filter functions in templates.

Reply:
I did this originally and it made the template look messy imo. Handling the filtering logic on the server side helped me write cleaner more easily-readable template logic.